tests/Unit/NumericalAlgorithms/LinearSolver/Test_Gmres.cpp

```diff
@@ -0,0 +1,244 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/DenseMatrix.hpp"
+#include "DataStructures/DenseVector.hpp"
+#include "Informer/Verbosity.hpp"
+#include "NumericalAlgorithms/Convergence/Criteria.hpp"
+#include "NumericalAlgorithms/Convergence/Reason.hpp"
+#include "NumericalAlgorithms/LinearSolver/Gmres.hpp"
+#include "Utilities/Gsl.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+namespace LinearSolver {
+namespace Serial {
+
+namespace {
+struct ScalarField : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+template <typename Tag>
+struct SomePrefix : db::PrefixTag {
+  using type = tmpl::type_from<Tag>;
+};
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.LinearSolver.Serial.Gmres",
+                  "[Unit][NumericalAlgorithms][LinearSolver]") {
+  {
+    /// [gmres_example]
+    INFO("Solve a symmetric 2x2 matrix");
+    DenseMatrix<double> matrix(2, 2);
+    matrix(0, 0) = 4.;
+    matrix(0, 1) = 1.;
+    matrix(1, 0) = 1.;
+    matrix(1, 1) = 3.;
+    const DenseVector<double> source{1., 2.};
+    const DenseVector<double> initial_guess{2., 1.};
+    const DenseVector<double> expected_solution{0.0909090909090909,
+                                                0.6363636363636364};
+    const Convergence::Criteria convergence_criteria{2, 1.e-14, 0.};
+    Gmres<DenseVector<double>> gmres{convergence_criteria,
```

Comment:
As @moxcodes commented on the old version, all these Gmres objects can be const.

Reply:
they are, perhaps GitHub didn't refresh something?